runtime/browser/lib/demo-base.js

```diff
@@ -59,33 +56,25 @@ class DemoBase extends HTMLElement {
     this.stageNo++;
     this.suggest();
   }
-  suggest() {
-    if (this.stage.recipes) {
-      this.suggestinator._getSuggestions = () => this.stage.recipes.map(
-          r => this.buildRecipe(r)
-        );
-      this.suggestinator
-        .suggestinate(this.arc)
-        .then(plans =>
-          plans.forEach((plan, i) => {
-            this.suggestions.add(plan, i);
-          })
-        );
+  async suggest() {
+    if (!this.stage.recipes) {
+      return;
     }
-  }
-  buildRecipe(info) {
-    let rb = new RecipeBuilder();
-    info.particles.forEach(pi => {
-      let p = rb.addParticle(pi.name);
-      if (pi.constrain) {
-        Object.keys(pi.constrain).forEach(k => {
-          p.connectConstraint(k, pi.constrain[k]);
-        });
-      }
+
+    let planner = new Planner();
+    // TODO: Shrug.. Context dependency between demo base and subclasses is weird.
+    this.context.recipes = this.stage.recipes;
+    planner.init(this.arc, this.context);
+    let plans = await planner.plan(500);
+    plans.forEach((plan, i) => {
+      // TODO: invoke speculator/decriptinator
+      const Speculator = require('../../speculator.js');
+      const DescriptionGenerator = require('../../description-generator.js');
+      let rank = 1;
+      let description = plan.toString();
+      //let description = plan.particles.map(p => p.spec.name).join(', '); //new DescriptoinGenerator();
```

Comment:
none of the views in the plan have IDs - is it expected? the description generator relies on views having unique ids.